force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,190 @@
+        // contacts.add(new Contact(FirstName='Warren', LastName='Walters', Email='walterstest@CCA.com'));
+        // contacts.add(new Contact(FirstName='Chase', LastName='BigDawg', Email='c.brammer@rcks.com'));
+        // contacts.add(new Contact(FirstName='Riley', LastName='Meyer'));
+
+        // Database.insert(contacts, AccessLevel.SYSTEM_MODE);
+
+        List<Lead> leads = new List<Lead>();
+        leads.add(new Lead(FirstName = 'David', LastName = 'Bolding', Company = 'Bolding Metals', Title = 'vp', Industry='Manufacturing', LeadSource='Web', Email='d.bolding@metals.com'));
+        leads.add(new Lead(FirstName = 'Warren', LastName = 'Walters', Company = 'CCA', Title = 'v.p.', Industry='Technology', LeadSource='Web', Phone='123-456-7890'));
+        leads.add(new Lead(FirstName = 'Chase', LastName = 'BigDawg', Company = 'BigDawg Rocks', Title = 'vice president', Industry='Construction', LeadSource='Purchased List', Email='c.bigdawg@rcks.com', Phone='456-654-9874'));
+        leads.add(new Lead(FirstName = 'Brooks', LastName = 'Wizard', Company = 'Wizard School', Title = 'mgr', Industry='Education', LeadSource='Purchased List', Email='b.magic@potions.com'));
+        leads.add(new Lead(FirstName = 'Oxana', LastName = 'Suvorova', Company = 'Suvorova Schools', Title = 'manage', Industry='Technology', LeadSource='Purchased List', Email='o.suvorova@schools.com'));
+        leads.add(new Lead(FirstName = 'John', LastName = 'George', Company = 'George Elite', Title = 'head of department', Industry='Apparel', LeadSource='Purchased List', Email='j.george@elite.com'));
+        leads.add(new Lead(FirstName = 'Navid', LastName = 'Mudassir', Company = 'M.U.D.', Title = 'exec', Industry='Technology', LeadSource='Web', Email='n.mudassir@mud.com'));
+        leads.add(new Lead(FirstName = 'Raghavendra', LastName = 'Yadavalli', Company = 'R.Y.', Title = 'chief', Industry='Utilities', LeadSource='Purchased List', Email='r.yadavalli@ry.com'));
+        leads.add(new Lead(FirstName = 'Tony', LastName = 'Kirumba', Company = 'Tony\'s Turtles', Title = 'head', Industry='Retail', LeadSource='Purchased List', Email='t.kirumba@turtlesrule.com'));
+        leads.add(new Lead(FirstName = 'Ariel', LastName = 'T.', Company = 'Under the Sea', Title = 'assist', Industry='Hospitality', LeadSource='Web', Email='a.t@underwater.com', Phone='654-321-5823'));
+        leads.add(new Lead(FirstName = 'Dallas', LastName = 'Bailey', Company = 'Bailey Tea', Title = 'deputy', Industry='Retail', LeadSource='Phone Inquiry', Email='d.bailey@teatime.com'));
+        leads.add(new Lead(FirstName = 'Tyler', LastName = 'Riden', Company = 'Rock n Riden', Title = 'jr', Industry='Construction', LeadSource='Partner Referral', Email='t.riden@rock4life.com'));
+        leads.add(new Lead(FirstName = 'Riley', LastName = 'Meyer', Company = 'Developer Futures', Title = 'developer', Industry='Technology', LeadSource='Web', Email='r.meyer@dev.com', Phone='987-654-3210'));
```

Comment:
It's a bit more advanced but you can use CSV to generate data https://trailhead.salesforce.com/content/learn/modules/unit-testing-on-the-lightning-platform/generate-data-for-tests

Reply:
It might help out with this complex data.

Reply:
You can also generate multiple sets of data using for loops and using modulus to change values like the title.